src/main/scala/cromwell/engine/finalcall/FinalCall.scala

```diff
@@ -34,4 +66,19 @@ trait FinalCall extends Scope {
   def prerequisiteScopes: Set[Scope] = Set.empty
 
   val parent: Option[Scope] = Option(this.rootWorkflow)
+
+  override def rootWorkflow: Workflow = workflow.namespace.workflow
+
+  override def unqualifiedName = FinalCall.name(workflow.name, companion.finalCallName)
+}
+
+trait FinalCallCompanion[A <: FinalCall] {
```

Comment:
Just like `GenericCompanion.newBuilder` or `GenericCompanion.empty`, it provides companion, singleton methods. I'll mention this in scaladoc.

The companion object for each `FinalCall` allows instantiation of the class from the singleton, and also includes a unique locally qualified name for each `FinalCall` type.